pkg/hvutils/hvutils.go

```diff
@@ -0,0 +1,50 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hvutils
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/validation"
+)
+
+func GetInstanceTypeFromTags(tags []string, deviceId int32) (string, error) {
+	prefix := "instance-type="
+	var instanceTypes []string
+	for _, tag := range tags {
+		if !strings.HasPrefix(tag, prefix) {
+			continue
+		}
+		instanceType := strings.TrimSpace(tag[len(prefix):])
+		instanceTypes = append(instanceTypes, instanceType)
+	}
+	if len(instanceTypes) == 0 {
+		return "", fmt.Errorf("No instance-type tags found on deviceId=%d", deviceId)
```

Comment:
Should we really return an error in this case? Is it not a "normal thing"? I guess we should rather return a boolean that indicates whether a type has been found or not.

Reply:
Since this will change, if HV provides Instance-Types in their API, I would leave it like it is at the moment.